### PR TITLE
Implement googlepay button

### PR DESCRIPTION
**Pull request by elizalucas**
This PR implements the following:

1. Displays the googlepay button for stg/prod to get payment data from the Google Pay API
2. Calls the convert token endpoint in wallets-api
3. Adds a 'tokens' option to the POST payments page

Note: this is only initial implementation of the google pay button - there are some hard coded values in the googlePay lib in this PR that will be removed/moved to SSM in subsequent PRs

**Sandbox/Smokebox:**
Screen Shot 2022-03-04 at 5 40 23 PM

**Stg/Prod:**
Screen Shot 2022-03-04 at 5 40 56 PM

### CODE REVIEW

File: pages/debug/payments/googlepay/convertToken.vue [hunk 1]
```diff
@@ -93,6 +104,23 @@ export default class ConvertToken extends Vue {
   showError = false
   payload = {}
 
+  buttonOptions: ButtonOptions = {
+    onClick: this.onGooglePayButtonClicked,
+    allowedPaymentMethods: [
+      {
+        type: 'CARD',
```
Comment (antiv0) [inline pages/debug/payments/googlepay/convertToken.vue:111]: nit: (to my memory)
when i did  these hardcoded config values i extracted it in separate config object, so later you have it one place

like you don't allow amex below, maybe you want to change that, maybe you want to change auth methods, having in one place

**Review by antiv0 — DISMISSED** [review]
lgtm

File: lib/googlePay.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      },
+    },
+  ],
+}
+
+const paymentDataRequest: PaymentDataRequest = {
+  apiVersion: 2,
+  apiVersionMinor: 0,
+  merchantInfo: {
+    merchantId: '12345678901234567890',
```
Comment (huaweigu) [inline lib/googlePay.ts:24]: I think we would want it to be [real merchant id and name](https://docs.google.com/document/d/1J5JUFkSu7AVNhMd8qPO-Uzx_kC_MxjAot5TrGvcZwXE/edit#heading=h.56scnum8vqyt) for stg/prod, will we store them in ssm in next PR?

File: lib/googlePay.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      type: 'CARD',
+      parameters: {
+        allowedAuthMethods: ['PAN_ONLY', 'CRYPTOGRAM_3DS'],
+        allowedCardNetworks: ['MASTERCARD', 'VISA'],
+      },
+      tokenizationSpecification: {
+        type: 'PAYMENT_GATEWAY',
+        parameters: {
+          gateway: 'checkoutltd',
+          gatewayMerchantId: 'YOUR_PUBLIC_KEY',
```
Comment (huaweigu) [inline lib/googlePay.ts:38]: okay I think we can put it in SSM as well

File: lib/googlePay.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        },
+      },
+    },
+  ],
+  transactionInfo: {
+    currencyCode: 'USD',
+    countryCode: 'US',
+    totalPriceStatus: 'FINAL',
+    totalPrice: '12.00',
```
Comment (huaweigu) [inline lib/googlePay.ts:47]: hmm, will we remove this hardcoded value in next PR?

Comment (elizalucas) [reply lib/googlePay.ts:47]: Yes, will be removing or moving all hardcoded values in the next PR, just had them here for testing purposes for now

File: lib/googlePay.ts (hunk 2 above)
Comment (elizalucas) [reply lib/googlePay.ts:72]: Yes, will store the merchant values and Checkout details in SSM for next PR

**Review by huaweigu — APPROVED** [review]

File: lib/googlePay.ts (hunk 2 above)
Comment (antiv0) [reply lib/googlePay.ts:50]: // just for Eliza, i don't think she ever stored secrets, secrets go to kubernetes db (which is likely etcd), not in aws ssm

**Review by antiv0 — APPROVED** [review]
lgtm to get code out

File: lib/googlePay.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+function onGooglePayLoaded(buttonOptions: ButtonOptions) {
+  paymentsClient = getGooglePaymentsClient()
+  paymentsClient
+    .isReadyToPay(getIsReadyToPayRequest())
+    .then(function (response: IsReadyToPayResponse) {
+      if (response.result) {
+        const button = paymentsClient.createButton(buttonOptions)
+        document.getElementById('google-pay-button')?.append(button)
```
Comment (antiv0) [inline lib/googlePay.ts:103]: nit: nowhere else in this codebase is DOM manipulated
somebody with good experience in our frontend stack could likely propose a way to do this that is in line with rest of codebase

like usage of Vue's v-if

File: lib/walletsApi.ts [hunk 6]
```diff
@@ -102,10 +102,30 @@ function getMasterWallet() {
...
+ * @param {String} type
+ * @param {Object} tokenData
+ * @param {UUID} idempotencyKey
+ */
+function convertToken(type: string, tokenData: Object) {
+  const url = '/v1/tokens'
+  const payload = {
+    type,
+    tokenData,
+    idempotencyKey: '394dffd9-e992-4b86-b3f3-0a242a44db48',
```
Comment (antiv0) [inline lib/walletsApi.ts:116]: why not have this be randomly generated in this PR ?

**Review by itsikcircle — APPROVED** [review]

File: lib/googlePay.ts (hunk 2 above)
Comment (timmy-circle) [reply lib/googlePay.ts:50]: curious on where/what the local secret is stored?

Comment (antiv0) [reply lib/googlePay.ts:50]: this is public repo, in private repo deployment is defined

**Comment by huaweigu** [comment]
Do you mind linking the story in the PR title? Thanks!

**Comment by antiv0** [comment]
> 

This is public repo, no internal info gets associated with this repo

File: lib/googlePay.ts (hunk 2 above)
Comment (huaweigu) [reply lib/googlePay.ts:50]: I was wrong, it should live in K8S secrets for sample apple private

**Pull request merged**